backend/objstore/objstore.go

```diff
@@ -0,0 +1,93 @@
+package objstore
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/caarlos0/env"
+	"github.com/minio/minio-go/v6"
+	"github.com/minio/minio-go/v6/pkg/credentials"
+	"github.com/pace/bricks/http/transport"
+	"github.com/pace/bricks/maintenance/health/servicehealthcheck"
+	"github.com/pace/bricks/maintenance/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type config struct {
+	Endpoint        string `env:"S3_ENDPOINT" envDefault:"s3.amazonaws.com"`
+	AccessKeyID     string `env:"S3_ACCESS_KEY_ID"`
+	Region          string `env:"S3_REGION" envDefault:"us-east-1"`
+	SecretAccessKey string `env:"S3_SECRET_ACCESS_KEY"`
+	UseSSL          bool   `env:"S3_USE_SSL"`
+
+	HealthCheckBucketName string        `env:"S3_HEALTH_CHECK_BUCKET_NAME" envDefault:"health-check"`
+	HealthCheckObjectName string        `env:"S3_HEALTH_CHECK_OBJECT_NAME" envDefault:"latest.log"`
+	HealthCheckResultTTL  time.Duration `env:"S3_HEALTH_CHECK_RESULT_TTL" envDefault:"2m"`
+}
+
+var cfg config
+
+func init() {
+	prometheus.MustRegister(paceObjStoreTotal)
+	prometheus.MustRegister(paceObjStoreFailed)
+	prometheus.MustRegister(paceObjStoreDurationSeconds)
+
+	// parse log config
+	err := env.Parse(&cfg)
+	if err != nil {
+		log.Fatalf("Failed to parse object storage environment: %v", err)
+	}
+
+	client, err := Client()
+	if err != nil {
+		log.Fatalf("Failed to create object storage client: %v", err)
+	}
+	servicehealthcheck.RegisterHealthCheck(&HealthCheck{
+		Client: client,
+	}, "objstore")
+
+	ok, err := client.BucketExists(cfg.HealthCheckBucketName)
+	if err != nil {
+		log.Warnf("Failed to create check for bucket: %v", err)
+	}
+	if !ok {
+		err := client.MakeBucket(cfg.HealthCheckBucketName, cfg.Region)
+		if err != nil {
+			log.Warnf("Failed to create bucket: %v", err)
+		}
+	}
+}
+
+// Client with environment based configuration
+func Client() (*minio.Client, error) {
+	return CustomClient(cfg.Endpoint, &minio.Options{
+		Secure:       cfg.UseSSL,
+		Region:       cfg.Region,
+		BucketLookup: minio.BucketLookupAuto,
+		Creds:        credentials.NewStaticV4(cfg.AccessKeyID, cfg.SecretAccessKey, ""),
+	})
+}
+
+// CustomClient with customized client
+func CustomClient(endpoint string, opts *minio.Options) (*minio.Client, error) {
+	client, err := minio.NewWithOptions(endpoint, opts)
+	if err != nil {
+		return nil, err
+	}
+	log.Logger().Info().Str("endpoint", endpoint).
+		Str("region", opts.Region).
+		Bool("ssl", opts.Secure).
+		Msg("S3 connection created")
+	client.SetCustomTransport(newCustomTransport(endpoint))
+	return client, nil
+}
+
+func newCustomTransport(endpoint string) http.RoundTripper {
+	return transport.NewDefaultTransportChain().Use(newMetricRoundTripper(endpoint))
+}
+
+func newMetricRoundTripper(endpoint string) *metricRoundTripper {
```

Comment:
The `newMetricRoundTripper` function should be near the `metricRoundTripper` type definition.